Runtime/Scripts/RTCStats.cs

```diff
@@ -0,0 +1,809 @@
+                case StatsMemberType.Uint32:
+                    return NativeMethods.StatsMemberGetUnsignedInt(self);
+                case StatsMemberType.Int64:
+                    return NativeMethods.StatsMemberGetLong(self);
+                case StatsMemberType.Uint64:
+                    return NativeMethods.StatsMemberGetUnsignedLong(self);
+                case StatsMemberType.Double:
+                    return NativeMethods.StatsMemberGetDouble(self);
+                case StatsMemberType.String:
+                    return NativeMethods.StatsMemberGetString(self).AsAnsiStringWithFreeMem();
+                case StatsMemberType.SequenceBool:
+                    return NativeMethods.StatsMemberGetBoolArray(self, ref length).AsBoolArray((int)length);
+                case StatsMemberType.SequenceInt32:
+                    return NativeMethods.StatsMemberGetIntArray(self, ref length).AsIntArray((int)length);
+                case StatsMemberType.SequenceUint32:
+                    return NativeMethods.StatsMemberGetUnsignedIntArray(self, ref length).AsUnsignedIntArray((int)length);
+                case StatsMemberType.SequenceInt64:
+                    return NativeMethods.StatsMemberGetLongArray(self, ref length).AsLongArray((int)length);
+                case StatsMemberType.SequenceUint64:
+                    return NativeMethods.StatsMemberGetUnsignedLongArray(self, ref length).AsUnsignedLongArray((int)length);
+                case StatsMemberType.SequenceDouble:
+                    return NativeMethods.StatsMemberGetDoubleArray(self, ref length).AsDoubleArray((int)length);
+                case StatsMemberType.SequenceString:
+                    return NativeMethods.StatsMemberGetStringArray(self, ref length).AsStringArray((int)length);
+                default:
+                    throw new ArgumentException();
+            }
+        }
+    }
+
+    public class RTCStats : IReadOnlyDictionary<string, object>
+    {
+        private IntPtr self;
+        internal Dictionary<string, RTCStatsMember> m_members;
+
+        public RTCStatsType Type
+        {
+            get
+            {
+                return NativeMethods.StatsGetType(self);
+            }
+        }
+
+        public string Id
+        {
+            get { return NativeMethods.StatsGetId(self).AsAnsiStringWithFreeMem(); }
+        }
+
+        public long Timestamp
+        {
+            get { return NativeMethods.StatsGetTimestamp(self); }
+        }
+
+        public bool ContainsKey(string key)
+        {
+            return m_members.ContainsKey(key);
+        }
+
+        public bool TryGetValue(string key, out object value)
+        {
+            RTCStatsMember member;
+            if (!m_members.TryGetValue(key, out member))
+            {
+                value = null;
+                return false;
+            }
+            value = member.GetValue();
+            return true;
+        }
+
+        public IEnumerable<string> Keys
```

Comment:
I think we will have less code if we can provide a function that returns m_members as IDictionary, so that the caller can call Keys, Values, etc